### ISSUE TITLE
Output dimension of Conv2dSubsampling

### ISSUE THREAD

**Issue by kan-bayashi** [espnet/espnet#822]
`Conv2dSubsampling` layer cause dimension mismatch when idim is multiple of 4.
https://github.com/espnet/espnet/blob/577baff5c5d229c860ca9936daba3cb51b9857cb/espnet/nets/pytorch_backend/transformer/subsampling.py#L14-L25

```python
[ins] In [8]: conv = torch.nn.Conv2d(1, 1, 3, 2)

[ins] In [9]: input = torch.zeros((1, 1, 20, 20))

[ins] In [10]: out1 = conv(input)

[ins] In [11]: out1.shape
Out[11]: torch.Size([1, 1, 9, 9])

[ins] In [12]: out2 = conv(out1)

[ins] In [13]: out2.shape
Out[13]: torch.Size([1, 1, 4, 4])
```

**Comment by Fhrozen** [comment]
let me fix this in #774

**Comment by Fhrozen** [comment]
Fixed in #774

**Comment by kan-bayashi** [comment]
Fixed #774 #822 #921
